src/libraries/Common/src/System/Security/Cryptography/MLKemImplementation.Windows.cs

```diff
@@ -0,0 +1,266 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Runtime.InteropServices;
+using System.Text;
+using Internal.NativeCrypto;
+using Microsoft.Win32.SafeHandles;
+
+using NTSTATUS = Interop.BCrypt.NTSTATUS;
+using KeyBlobMagicNumber = Interop.BCrypt.KeyBlobMagicNumber;
+using KeyBlobType = Interop.BCrypt.KeyBlobType;
+using BCRYPT_MLKEM_KEY_BLOB = Interop.BCrypt.BCRYPT_MLKEM_KEY_BLOB;
+
+namespace System.Security.Cryptography
+{
+    internal sealed partial class MLKemImplementation : MLKem
+    {
+        private static readonly SafeBCryptAlgorithmHandle? s_algHandle = OpenAlgorithmHandle();
+        private readonly bool _hasSeed;
+        private readonly bool _hasDecapsulationKey;
+        private SafeBCryptKeyHandle _key;
+
+        [MemberNotNullWhen(true, nameof(s_algHandle))]
+        internal static new bool IsSupported =>
+#if NET || NETSTANDARD2_0_OR_GREATER
+            // MLKemImplementation.Windows is used for all platforms in Microsoft.Bcl.Cryptography, so it checks that the
+            // platform is Windows.
+            RuntimeInformation.IsOSPlatform(OSPlatform.Windows) && s_algHandle is not null;
```

Comment:
Isn't the platform check redundant?  Non-Windows sets `s_algHandle` to null...

Reply:
Ah, yeah. First I fixed this method, then I fixed it in `OpenAlgorithmHandle`, but didn't go back to the first.